src/predicates/propPathSatisfies.js

```diff
@@ -0,0 +1,57 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Ian Hofmann-Hicks (evilsoft) */
+
+const curry = require('../core/curry')
+const isArray = require('../core/isArray')
+const isDefined = require('../core/isDefined')
+const isEmpty  = require('../core/isEmpty')
+const isInteger = require('../core/isInteger')
+const isNil = require('../core/isNil')
+const isPredOrFunc = require('../core/isPredOrFunc')
+const isString = require('../core/isString')
+const predOrFunc = require('../core/predOrFunc')
+
+// propPathSatisfies: [ (String | Integer) ] -> (a -> Boolean) -> b -> Boolean
+// propPathSatisfies: [ (String | Integer) ] -> Pred a -> b -> Boolean
+function propPathSatisfies(keys, pred, x) {
+  if(!isArray(keys)) {
+    throw new TypeError(
+      'propPathSatisfies: Array of Non-empty Strings or Integers required for first argument'
+    )
+  }
+
+  if(!isPredOrFunc(pred)) {
+    throw new TypeError(
+      'propPathSatisfies: Pred or predicate function required for second argument'
+    )
+  }
+
+  if(isNil(x)) {
+    return false
+  }
+
+  let target = x
+  for(let i = 0; i < keys.length; i++) {
+    const key = keys[i]
+
+    if(!(isString(key) && !isEmpty(key) || isInteger(key))) {
+      throw new TypeError(
+        'propPathSatisfies: Array of Non-empty Strings or Integers required for first argument'
+      )
+    }
+
+    if(isNil(target)) {
+      return false
+    }
+
+    target = target[key]
+
+    if(!isDefined(target)) {
```

Comment:
Should we really be defaulting to `false` in such cases? 

I understand that it's an edge case, but there are certain libs (mainly ORMs) that treat `undefined` as a legit value. The same goes for `hasPropPath` - should we consider piping `undefined` to the pred instead?

Reply:
Okay, after fiddling with ways to get this to make sense, there is no way to tell if the key exists with an `undefined` value when the path is the same depth of the target path. This was the same problem we ran into with the other path functions.

@karthikiyengar
I say we keep it in until people start reporting their specific edge cases with the specific libs they are using and address them on a case-by-case basis. we could check `Object.keys` but I do not want to add extra cycles until a lot of users need it.

Reply:
How about this before `target = target[key]`?

```
if (target.hasOwnProperty(key) && !target[key]) {
  return !!predOrFunc(pred, target[key])
}
```